src/key.rs

```diff
@@ -35,7 +101,7 @@ impl From<SignedSecretKey> for Key {
 impl std::convert::TryFrom<Key> for SignedSecretKey {
     type Error = ();
 
-    fn try_from(value: Key) -> Result<Self, Self::Error> {
+    fn try_from(value: Key) -> std::result::Result<Self, Self::Error> {
```

Comment:
You can probably override `Result` type inside `SignedSecretKey` and use `Self::Result`.

Reply:
Not sure I understand this, you mean add `type Result = std::result::Result` on line 107 just below `type Error = ()`?  Are we allowed to add new associated types to a trait in an impl?  That would surprise me.  Also not really sure it would become more readable.